service/src/main/java/bio/terra/pipelines/service/ImputationService.java

```diff
@@ -151,14 +185,78 @@ public List<ListAppResponse> queryForWorkspaceApps() {
+      //              .methodUrl(pipeline.getWdlUrl())
+      //              .methodVersion("1.0");
+      //      logger.info(
+      //          "this is creating a new method in cbas: {}",
+      //          cbasService.createMethod(
+      //              cbasUri, samService.getTspsServiceAccountToken(), postMethodRequest));
+
+      MethodListResponse methodListResponse =
+          cbasService.getAllMethods(cbasUri, samService.getTspsServiceAccountToken());
+      logger.info("list of methods available: {}", methodListResponse);
+
+      // grab methodVersionId needed to submit a submission
+      UUID methodVersionId = null;
+      for (MethodDetails methodDetails : methodListResponse.getMethods()) {
+        if (methodDetails.getName().equals(pipeline.getWdlMethodName())) {
+          methodVersionId = methodDetails.getMethodVersions().get(0).getMethodVersionId();
+          logger.info("this is the method version id: {}", methodVersionId);
+          break;
+        }
+      }
+
+      // launch a cbas submission
+      RunSetRequest runSetRequest =
+          new RunSetRequest()
+              .runSetDescription("run set description")
+              .runSetName("run set name")
+              .methodVersionId(methodVersionId)
+              .addWorkflowInputDefinitionsItem(
+                  new WorkflowInputDefinition()
+                      .inputName("HelloWorld.scatter_num")
+                      .inputType(
+                          new ParameterTypeDefinitionPrimitive()
+                              .primitiveType(PrimitiveParameterValueType.INT)
+                              .type(ParameterTypeDefinition.TypeEnum.PRIMITIVE))
+                      .source(
+                          new ParameterDefinitionRecordLookup()
+                              .recordAttribute("scatter")
+                              .type(ParameterDefinition.TypeEnum.RECORD_LOOKUP)))
+              .addWorkflowOutputDefinitionsItem(
+                  new WorkflowOutputDefinition()
+                      .outputName("HelloWorld.output_file")
+                      .outputType(
+                          new ParameterTypeDefinitionArray()
+                              .nonEmpty(true)
+                              .arrayType(
+                                  new ParameterTypeDefinitionPrimitive()
+                                      .primitiveType(PrimitiveParameterValueType.FILE)
+                                      .type(ParameterTypeDefinition.TypeEnum.PRIMITIVE))
+                              .type(ParameterTypeDefinition.TypeEnum.ARRAY))
+                      .destination(
+                          new OutputDestinationRecordUpdate()
+                              .recordAttribute("output_file")
+                              .type(OutputDestination.TypeEnum.RECORD_UPDATE)))
+              .wdsRecords(new WdsRecordSet().recordType(wdsTableName).addRecordIdsItem(id));
       logger.info(
```

Comment:
is it normal to have code executed inside a logger command?

Reply:
if you're not going to do anything with what the function returns except print it out then its fine